### PR TITLE
Merkle Proof verification feature for NFT claim

### PR DESCRIPTION
**Pull request by h4n0**
There are 2 parts of this PR:
1. In NFT pallet, Merkle Proof verification has been implemented to verify if user's address is in whitelist and mint NFT to user if it is in the list.
2. Add TypeScript distributor demo to generate Merkle Tree and Merkle Proof to certain user.

Details of design will be documented in notion.

### CODE REVIEW

File: pallets/nft/src/lib.rs [hunk 1]
```diff
@@ -66,6 +69,18 @@ pub struct ClassData<BN, ID> {
 	pub end_block: Option<BN>,
 	/// merged from two class; if true, burn the two items 
 	pub class_type: ClassType<ID>,
+  /// claimed vec for claim type NFT class, to guarantee each user claims once
```
Comment (geastwood) [inline pallets/nft/src/lib.rs]: Identation seems to be off.

File: pallets/nft/src/lib.rs [hunk 2]
```diff
@@ -66,6 +69,18 @@ pub struct ClassData<BN, ID> {
...
+  /// claimed vec for claim type NFT class, to guarantee each user claims once 
+  claimed_vec: Vec<u32>, // maximal index of claiming user is 2^32 which is more than enough
+}
+
+impl<BN, ID> ClassData<BN, ID> {
+  fn is_claimed(&self, index: u32) -> bool {
+    self.claimed_vec.contains(&index)
+  }
+
+  fn record_claim(&mut self, index: u32) {
```
Comment (geastwood) [inline pallets/nft/src/lib.rs]: Don't we check wether this is already claimed? It seems too easy to just `push` the index.

**Comment by geastwood** [comment]
Why this PR has the changes from @Satoshi-Kusumoto, can we change the base branch so it's possible to review the changes?

File: pallets/nft/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+							Error::<T>::NonBurnable
+						);
+						let data2 = class_info2.data;
+						ensure!(
+							data2.properties.0.contains(ClassProperty::Burnable),
+							Error::<T>::NonBurnable
+						);
+					}
+				}
+        ClassType::Claim(_) => {
```
Comment (suinuj) [inline pallets/nft/src/lib.rs]: indent issue

**Comment by h4n0** [comment]
I will do a rebase and create another PR to avoid this github auto-rebase chaos.

**Comment by h4n0** [comment]
Also fix the indentation issue, thanks for the reviews.

**Comment by h4n0** [comment]
Closed. Resubmitted with #23.

**Pull request closed without merging**